arrows/core/video_input_image_list.cxx

```diff
@@ -233,31 +244,39 @@ ::end_of_video() const
 }
 
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 bool
 video_input_image_list
 ::good() const
 {
   return d->m_frame_number > 0 && ! this->end_of_video();
 }
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 bool
 video_input_image_list
 ::seekable() const
 {
   return true;
 }
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 size_t
 video_input_image_list
 ::num_frames() const
 {
   return d->m_files.size();
 }
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+kwiver::vital::path_t
+video_input_image_list
+::filename() const
+{
+  return *(d->m_current_file);
```

Comment:
This seems inconsistent to me.  the "filename" of a video, if it exists, should have similar meaning for encoded videos and for image sequences.  For an encoded video this would be something like "videoname.mpg" and would be fixed over the duration of the video.  For an image list I would expect it to be the directory, glob, or image list text file specifying which frames to load.

You can already access the filename of the current frame through the metadata object associated with the current frame.